### PR TITLE
Update balloon test to use fragments, disable back button, and add behavior for onPause()/onResume()

### PR DESCRIPTION
**Pull request by skoy7**
layout might be a bit off

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
```
Comment (ccharyi) [inline build.gradle:8]: Was there a particular reason that you changed the build dependencies for the project? It worked fine with 2.2.3.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Mon Dec 28 10:00:20 PST 2015
+#Sat Apr 08 00:13:41 EDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (ccharyi) [inline gradle/wrapper/gradle-wrapper.properties:6]: Same here, did this change by updating your plugins?

**Review by ccharyi — COMMENTED** [review]
Functionality wise it works fine as before, I'm not entirely sure that you needed to change the gradle build dependencies (unless android studio did that for you). The test seems to be letterboxed a bit, wasn't sure if that was expected behavior. Good work!

**Comment by skoy7** [comment]
Yeah, that was unintentional, I didn't manually change those files. I'll
revert and merge when I get home tonight.

**Comment by skoy7** [comment]
Also, I'm not quite sure I know what you mean by the test seeming
"letterboxed"

**Comment by ccharyi** [comment]
Actually, you probably don't need to revert the gradle files back, I think it happened because you updated your gradle plugin. So long as everyone else also updates their plugin after the merge we shouldn't have any problems.

The test has some weird padding going on, seen here:
image

I noticed that the balloon test didn't have this previously, not sure what caused it exactly

**Comment by skoy7** [comment]
Ah yeah, I did notice the layout was being weird, but it was late and I
couldn't see what went wrong. I'll look more into it tonight

**Pull request merged**